### PR TITLE
Add helper functions for loading/dumping to buffers and file descriptors

### PR DESCRIPTION
**Pull request by npmccallum**

### CODE REVIEW

File: doc/apiref.rst [hunk 1]
```diff
@@ -1120,6 +1157,33 @@ If no error or position information is needed, you can pass *NULL*.
...
+   above.
+
+   This function will start reading the input from whatever position
+   the input file descriptor was in, without attempting to seek first.
+   If an error occurs, the file position will be left indeterminate.
+   On success, the file position will be at EOF, unless
+   ``JSON_DISABLE_EOF_CHECK`` flag was used. In this case, the file
+   descriptor's position will be at the first character after the last
+   ``]`` or ``}`` in the JSON input. This allows calling
+   :func:`json_loadf()` on the same file descriptor multiple times,
```
Comment (akheron) [inline doc/apiref.rst]: Should be `json_loadfd()`

**Comment by akheron** [comment]
It seems that `unistd.h` doesn't exist on Windows. IIRC, Windows has file descriptors and equivalents of `read()` and `write()`, but I don't know how to use them.

File: doc/apiref.rst (hunk 1 above)
Comment (npmccallum) [inline doc/apiref.rst]: Fixed in the latest commit.

**Comment by npmccallum** [comment]
I can't seem to make the automated tests run again for some reason (and I'm trying to get a test to see if I've fixed building on win32). I'm going to close this PR and make a new one.

**Pull request closed without merging**